tests/v2/test_waku_filter.nim

```diff
@@ -5,213 +5,191 @@ import
   testutils/unittests,
   chronos, 
   chronicles,
-  libp2p/crypto/crypto,
-  libp2p/multistream
+  libp2p/crypto/crypto
 import
   ../../waku/v2/node/peer_manager/peer_manager,
   ../../waku/v2/protocol/waku_message,
   ../../waku/v2/protocol/waku_filter,
-  ../test_helpers, 
+  ../../waku/v2/protocol/waku_filter/client,
   ./utils,
   ./testlib/common,
   ./testlib/switch
 
 
-const dummyHandler = proc(requestId: string, msg: MessagePush) {.async, gcsafe, closure.} = discard
+proc newTestWakuFilterNode(switch: Switch, timeout: Duration = 2.hours): Future[WakuFilter] {.async.} =
+  let
+    peerManager = PeerManager.new(switch)
+    rng = crypto.newRng()
+    proto = WakuFilter.new(peerManager, rng, timeout)
 
+  await proto.start()
+  switch.mount(proto)
+
+  return proto
+
+proc newTestWakuFilterClient(switch: Switch): Future[WakuFilterClient] {.async.} =
+  let
+    peerManager = PeerManager.new(switch)
+    rng = crypto.newRng()
+    proto = WakuFilterClient.new(peerManager, rng)
+
+  await proto.start()
+  switch.mount(proto)
+
+  return proto
 
-# TODO: Extend test coverage
-procSuite "Waku Filter":
 
+# TODO: Extend test coverage
+suite "Waku Filter":
   asyncTest "should forward messages to client after subscribed":
     ## Setup
-    let rng = crypto.newRng()
     let 
-      clientSwitch = newTestSwitch()
       serverSwitch = newTestSwitch()
+      clientSwitch = newTestSwitch()
 
     await allFutures(serverSwitch.start(), clientSwitch.start())
 
-    ## Given
-    # Server
-    let
-      serverPeerManager = PeerManager.new(serverSwitch) 
-      serverProto = WakuFilter.init(serverPeerManager, rng, dummyHandler)
-    await serverProto.start()
-    serverSwitch.mount(serverProto)
-
-    # Client
-    let handlerFuture = newFuture[(string, MessagePush)]()
-    proc handler(requestId: string, push: MessagePush) {.async, gcsafe, closure.} =
-      handlerFuture.complete((requestId, push))
+    let 

```

Comment:
In general, perhaps you could reuse more code across the tests? move some stuff to a common place? I see some boilerplate code that makes the tests more difficult to read and distracts from whats being tested.

Reply:
Can you be more specific? I don't understand what you are calling "boilerplate code"

Reply:
the first 30-ish lines of each test case look almost identical (`timeout=200.milliseconds` variation in some cases). wondering if we could move more code to a common function, just as you do with `newTestWakuFilterNode`. just for the sake of readability.